### PR TITLE
MAINT: Prefer PySide6 in testing

### PR DESCRIPTION
**Pull request by larsoner**
~~I think we should cut a release. Given that this will be the first to support Qt6, I bumped to 0.3.0 (last release was 0.2.6).~~

~~@agramfort or @hoechenberger feel free to merge and I'll cut a release~~

Uses PySide6 as our preferred framework in testing. We can't change `environment.yml` until conda-forge manages to release PySide6, which will require Qt6 first. So it might be a while!

Todo

- [x] Fix bug with mne_dspm example (error, or hanging ref?)

### CODE REVIEW

**Comment by larsoner** [comment]
Argh, I meant for that to be the text for the mne-qt-browser PR I'm about to open. This PR might require a 0.3.0 release, we'll see!

**Comment by larsoner** [comment]
PySide6's wheel does not work on 18.04 LTS, and switching to 20.04 LTS we get volume rendering bugs from MESA

| main | PR |
| -- | -- |
image | image

I'll try PyQt6 on CircleCI + Ubuntu 18.04. That might be a way forward at least...

**Comment by larsoner** [comment]
Okay this one is good to go. We still use PyQt5 on CircleCI because PySide6 uses too new an ABI for 18.04, and we can't move away from 18.04 until https://gitlab.freedesktop.org/mesa/mesa/-/issues/6309 is sorted out

File: mne/viz/_3d.py [hunk 1]
```diff
@@ -3292,24 +3292,28 @@ def plot_brain_colorbar(ax, clim, colormap='auto', transparent=True,
 
 @dataclass()
 class _3d_Options:
-    antialias: Optional[str]
-    depth_peeling: Optional[str]
-    smooth_shading: Optional[str]
+    antialias: Optional[bool]
+    depth_peeling: Optional[bool]
+    smooth_shading: Optional[bool]
+    multi_samples: Optional[int | None]
```
Comment (larsoner) [inline mne/viz/_3d.py:3298]:
```suggestion
    multi_samples: Optional[int]
```

**Comment by larsoner** [comment]
Now the problem with PySide6 on CircleCI is a remaining reference:

https://app.circleci.com/pipelines/github/mne-tools/mne-python/13724/workflows/cffe0e33-f535-43fc-8b0f-1217939638a8/jobs/43496

I'll take care of that in a separate PR, though. Ready for review/merge from my end!

File: azure-pipelines.yml [hunk 2]
```diff
@@ -257,9 +257,9 @@ stages:
           PLATFORM: 'x86-64'
           TEST_MODE: 'conda'
           PYTHON_VERSION: '3.9'
-        3.7 pip:
+        3.9 pip:
           TEST_MODE: 'pip'
-          PYTHON_VERSION: '3.7'
+          PYTHON_VERSION: '3.9'
```
Comment (hoechenberger) [inline azure-pipelines.yml:262]: Can we maybe switch to 3.10 directly?

File: doc/install/advanced.rst [hunk 3]
```diff
@@ -191,6 +191,11 @@ to force MESA to use modern OpenGL by using this before executing
 
 Also, it's possible that different software rending backends might perform
 better than others, such as using the ``llvmpipe`` backend rather than ``swr``.
+In newer MESA (21+), rendering is off when using MSAA, so consider setting:
```
Comment (hoechenberger) [inline doc/install/advanced.rst:194]: "off" as in "faulty"? Could this be rephrased to be more clear?

File: doc/changes/latest.inc [hunk 4]
```diff
@@ -33,6 +33,8 @@ Bugs
 
 - Fix bug where ``theme`` was not handled properly in :meth:`mne.io.Raw.plot` (:gh:`10487`, :gh:`10500` by `Mathieu Scheltienne`_ and `Eric Larson`_)
 
+- Fix bug with newer MESA rendering made possible via ``MNE_3D_OPTION_MULTI_SAMPLES=1`` env var (:gh:`10513` by `Eric Larson`_)
```
Comment (hoechenberger) [inline doc/changes/latest.inc:36]: Just reading this message, I didn't realize this variable was a **new** thing. How about:

```suggestion
- Rendering issues with recent MESA releases can be avoided by setting the new environment variable``MNE_3D_OPTION_MULTI_SAMPLES=1`` (:gh:`10513` by `Eric Larson`_)
```

Comment (hoechenberger) [reply doc/changes/latest.inc:36]: But should probably also mention the new kwarg of `set_3d_options()`

File: azure-pipelines.yml (hunk 2 above)
Comment (larsoner) [reply azure-pipelines.yml:262]: Not easily until VTK makes a wheel. They should release 9.2 soon, which will have one

**Comment by larsoner** [comment]
Okay locally I was able to get PySide6 working by making appropriate use of `weakref`. Hopefully it comes back green!

@hoechenberger I addressed the comments above, so I'll merge if it does indeed come back green.

**Review by hoechenberger — COMMENTED** [review]
LGTM! Thanks @larsoner, feel free to merge once green

**Pull request merged**